### ISSUE TITLE
getRegisteredUsersFromAddressBook returns only phone field. and full_name returns null.

### ISSUE THREAD

**Issue by BhavyKoshti9spl** [ConnectyCube/connectycube-flutter-sdk-releases#23]
BODY
  {"items":[{"user":{"id":2866889,"phone":"***********","address_book_name":"*******"}}]}

this type of body i am getting while calling getRegisteredUsersFromAddressBook()

and returned CubeUser list element will looks like this..

{full_name: null, email: null, login: null, phone: ***********, website: null, last_request_at: null, external_user_id: null, facebook_id: null, twitter_id: null, password: null, oldPassword: null, custom_data: null, avatar: null, tag_list: null, id: 2866889, created_at: null, updated_at: null}

why is full_name is empty ?

**Comment by TatankaConCube** [comment]
Looks like your users don't have `full_name` field. Do you add it on `signUp` step? If you use Firebase Phone Auth, you can add one more step after phone verification for setting the user's full name. You can do this using our API `updateUser(CubeUser user)`.

**Comment by BhavyKoshti9spl** [comment]
i can see the full name from the dashboard. but it is null in the getRegisteredUsersFromAddressBook() 

steps to reproduce : 
1. signin to the user.
2. fetch phonebook and update it on the server.
3. get the registered user from the server.

what i think is issue is in your mapping see the response from the server the key name is "address_book_name" and CubeUser param name is full_name.

**Comment by TatankaConCube** [comment]
can you, please provide the full log from the flutter console with request and response. Additionally, please try to get the user by id `2866889` via `getUserById(int id)`, and provide the log too.

**Comment by BhavyKoshti9spl** [comment]
response of getUserById is :

```
HEADERS
  {connection: keep-alive, cache-control: max-age=0, private, must-revalidate, set-cookie: _mkra_ctxt=a99c49ae6e49d0a8e8feb9ed6830536a--200; path=/; max-age=5; HttpOnly; secure, status: 200 OK, transfer-encoding: chunked, date: Mon, 07 Dec 2020 11:51:53 GMT, access-control-allow-origin: *, strict-transport-security: max-age=31536000,max-age=15768000;, content-type: application/json; charset=utf-8, x-xss-protection: 1; mode=block, server: nginx/1.16.1, x-request-id: 1a1ea6fc-39e1-458c-b193-29ef29f0f9c1, cb-token-expirationdate: 2020-12-07 13:51:45 UTC, connectycube-rest-api-version: 0.1.1, x-runtime: 0.036087, etag: W/"2d8a27074ade99a74e33532f6aa87d44", x-frame-options: SAMEORIGIN, x-content-type-options: nosniff}
BODY
  {"user":{"id":2866889,"full_name":"Prabhat Koshti","email":null,"login":"9428613430","phone":"9428613430","website":null,"created_at":"2020-12-02T06:25:41Z","updated_at":"2020-12-07T05:57:06Z","last_request_at":"2020-12-07T04:43:51Z","external_user_id":null,"facebook_id":null,"twitter_id":null,"blob_id":null,"custom_data":"","avatar":"","user_tags":null}}
```

and the full log for the error is : 

```
CB-SDK: : =========================================================
=== REQUEST ==== 1923e3f0-4a21-4867-b4f3-d6b06b870f37 ===
REQUEST
  GET https://api.connectycube.com/address_book/registered_users?udid=c2ca194854fb1884&compact=1
HEADERS
  {Content-type: application/json, ConnectyCube-REST-API-Version: 0.1.1, CB-SDK: Flutter 1.0.0, CB-Token: db76f44946aceb7161f5e9840f06b7fb3c000ef6}
BODY
 
 *** RESPONSE *** 200 *** 1923e3f0-4a21-4867-b4f3-d6b06b870f37 ***
HEADERS
  {connection: keep-alive, cache-control: max-age=0, private, must-revalidate, set-cookie: _mkra_ctxt=781afc05baaefbabf0e80aed2437715f--200; path=/; max-age=5; HttpOnly; secure, status: 200 OK, transfer-encoding: chunked, date: Mon, 07 Dec 2020 11:54:51 GMT, access-control-allow-origin: *, strict-transport-security: max-age=31536000,max-age=15768000;, content-type: application/json; charset=utf-8, x-xss-protection: 1; mode=block, server: nginx/1.16.1, x-request-id: 37576b91-b3c6-4739-b4b5-743c9231e1c8, cb-token-expirationdate: 2020-12-07 13:51:44 UTC, connectycube-rest-api-version: 0.1.1, x-runtime: 0.045570, etag: W/"c12f1e37be7849cb5f4465b784ab6b44", x-frame-options: SAMEORIGIN, x-content-type-options: nosniff}
BODY
  {"items":[{"user":{"id":2866889,"phone":"9428613430","address_book_name":"Prabhat G. Kos"}}]}

```
and when i print the element of the list which i got in return then it will look like this..

`{full_name: null, email: null, login: null, phone: 9428613430, website: null, last_request_at: null, external_user_id: null, facebook_id: null, twitter_id: null, password: null, oldPassword: null, custom_data: null, avatar: null, tag_list: null, id: 2866889, created_at: null, updated_at: null}`

my code is as follows :

```
  void fetchFriendsList() async {
    getRegisteredUsersFromAddressBook(true, await _getId()).then((usersList) {
      _isUsersContinues = false;
      userToSearch = null;
      userMsg = " ";
      userList.clear();
      
      usersList.forEach((element) {
        print(element);
      });
      setState(() {
        this.userList.addAll(usersList);
        print("USER LIST ${userList.length}");
      });
    }).catchError((error) {});
  }
```

if anything else you want please let me know i will add

**Comment by TatankaConCube** [comment]
@BhavyKoshti9spl try to load users in full view not compact, for it call `getRegisteredUsersFromAddressBook(false, await _getId())`. According to [our documentation](https://developers.connectycube.com/flutter/address-book?id=retrieve-registered-users), for `compact` view you will receive only user's ID and phone.

**Comment by BhavyKoshti9spl** [comment]
@TatankaConCube  you guys are amazing thanks a lot. and please close this issue.

**Comment by TatankaConCube** [comment]
hello @BhavyKoshti9spl, today we released a new version ([1.1.0](https://pub.dev/packages/connectycube_sdk/changelog)) of our SDK, and now on request `getRegisteredUsersFromAddressBook` in the compact mode you will receive users with field `addressBookName`. In this field, you will receive the name of your user from your uploaded address book. 

If your issue resolved, please close this ticket.

**Comment by TatankaConCube** [comment]
No activity for a long time, closing.
